src/api_manager/service_control/proto.cc

```diff
@@ -1402,7 +1402,14 @@ Status Proto::ConvertCheckResponse(const CheckResponse& check_response,
   // TODO: report a detailed status to the producer project, but hide it from
   // consumer
   // TODO: unless they are the same entity
+
+  // If status() is available, use it.
   const CheckError& error = check_response.check_errors(0);
+  if (error.has_status()) {
```

Comment:
Are you sure about this? I think it breaks fail-open on line 1498

Reply:
You are right.  I will remove it.